dockerunit-core/src/main/java/com/github/qzagarese/dockerunit/DockerUnitRule.java

```diff
@@ -0,0 +1,135 @@
+
+            @Override
+            public void evaluate() throws Throwable {
+                
+                logger.log(Level.INFO, "Performing service discovery for the following context: " + description.getDisplayName());
+                try {
+                    doSetup(description);
+                    base.evaluate(); // This will run the test.
+                } finally {
+                    logger.log(Level.INFO, "Cleaning up active services for the following context: " + description.getDisplayName());
+                    doTeardown();
+                }
+            }
+        };
+    }
+
+    private void doSetup(final Description description) {
+        UsageDescriptor descriptor = descriptorBuilder.buildDescriptor(description.getTestClass());
+        UsageDescriptor discoveryProviderDescriptor = descriptorBuilder.buildDescriptor(discoveryProvider.getDiscoveryConfig());
+      
+        // Build discovery context
+        ServiceContext discoveryContext = contextBuilder.buildContext(discoveryProviderDescriptor);
+        DockerUnitRule.this.discoveryContext = discoveryContext;
+        if(!discoveryContext.allHealthy()) {
+            throw new RuntimeException(discoveryContext.getFormattedErrors());
+        }
+
+        
+        // Create containers and perform discovery one service at the time
+        List<ServiceContext> serviceContexts = new ArrayList<>();
+        descriptor.getDependencies().forEach(usage -> {
+            ServiceContext context = contextBuilder.buildServiceContext(usage);
+            if(!context.allHealthy()) {
+                throw new RuntimeException(context.getFormattedErrors());
+            }
+            context = discoveryProvider.populateRegistry(context);
+            serviceContexts.add(context);
+        });
+        ServiceContext completeContext = mergeContexts(serviceContexts);
+        activeContexts.put(DockerUnitRule.this.serviceContextName, completeContext);
+        if(!completeContext.allHealthy()) {
+            throw new RuntimeException(completeContext.getFormattedErrors());
+        }
+        
+    }
+
+    private ServiceContext mergeContexts(List<ServiceContext> serviceContexts) {
+        ServiceContext completeContext = null;
+        if (serviceContexts.size() > 0) {
+            completeContext = serviceContexts.remove(0);
+        }
+        for (ServiceContext serviceContext : serviceContexts) {
+            completeContext = completeContext.merge(serviceContext);
+        }
+        return completeContext;
+    }
+
+    private void doTeardown() {
+        ServiceContext context = activeContexts.get(DockerUnitRule.this.serviceContextName);
+        if(context != null) {
+            ServiceContext cleared = contextBuilder.clearContext(context);
+            discoveryProvider.clearRegistry(cleared, new DefaultServiceContext(new HashSet<>()));
+        }
+        
+        ServiceContext discoveryContext = DockerUnitRule.this.discoveryContext;
```

Comment:
why do this assignment? why not just do `if (discoveryContext != null)` (where `discoveryContext` is the member var)